syntax = "proto3";
package protos.playbooks;

import "protos/base.proto";
import "google/protobuf/wrappers.proto";
import "google/protobuf/struct.proto";


message PlaybookApiCallTask {
  enum Method {
    UNKNOWN = 0;
    GET = 1;
    POST = 2;
    PUT = 3;
    PATCH = 4;
    DELETE = 5;
  }
  Method method = 1;
  google.protobuf.StringValue url = 2;
  google.protobuf.Struct headers = 3;
  google.protobuf.Struct payload = 4;
  google.protobuf.UInt64Value timeout = 5;
  google.protobuf.Struct cookies = 6;
}